aws-sdk-core/lib/aws-sdk-core/credential_provider_chain.rb

```diff
@@ -54,7 +55,8 @@ def envar(keys)
       nil
     end
 
-    def shared_credentials(options = {})
+    def shared_credentials(options)
+      return nil if options[:config].nil?
```

Comment:
The default `options` parameter on this method would have raised a `NoMethodError` and it's a private method, so I got rid of the default.